inline_scan.sh

```diff
@@ -328,31 +336,40 @@ post_analysis() {
 	    fi
 	    exit 1
 	fi
-
-	get_scan_result_by_sha
 }
 
+# This is done instead of the -g option, as we want to tie the RepoDigest value present in the image
+# with the image id as much as possible, instead of generating our own digest or via skopeo.
 get_repo_digest_id() {
     # Check to see if repo digest exists
-    if [[ "${SYSDIG_IMAGE_ID}" == '123456890abcdefg' ]]; then
+    if [[ "${SYSDIG_IMAGE_DIGEST}" == 'sha256:123456890abcdefg' ]]; then
```

Comment:
Since you're calling this only when there's not digest set I think the check is not needed.

If you want to keep the check though, you need to extend it to cover all the function, so no calculation is done if `SYSDIG_IMAGE_DIGEST` is already set.